src/cdk-new/get-stage-options.ts

```diff
@@ -0,0 +1,89 @@
+import {join} from 'path';
+import type {
+  MethodDeploymentOptions,
+  StageOptions,
+} from 'aws-cdk-lib/aws-apigateway';
+import {
+  AccessLogFormat,
+  LogGroupLogDestination,
+  MethodLoggingLevel,
+} from 'aws-cdk-lib/aws-apigateway';
+import type {ILogGroup} from 'aws-cdk-lib/aws-logs';
+import {Duration} from 'aws-cdk-lib/core';
+
+export interface StageOptionsInit {
+  readonly accessLogGroup: ILogGroup | undefined;
+  readonly methodDeployments: readonly MethodDeployment[] | undefined;
+  readonly throttling: Throttling | undefined;
+  readonly loggingEnabled: boolean | undefined;
+  readonly metricsEnabled: boolean | undefined;
+  readonly tracingEnabled: boolean | undefined;
+}
+
+export interface MethodDeployment {
+  readonly httpMethod: string;
+  readonly publicPath: string;
+  readonly proxyName: string | undefined;
+  readonly cacheTtlInSeconds: number | undefined;
+}
+
+export interface Throttling {
+  readonly burstLimit: number;
+  readonly rateLimit: number;
+}
+
+export function getStageOptions(init: StageOptionsInit): StageOptions {
+  const {
+    accessLogGroup,
+    methodDeployments,
+    throttling,
+    loggingEnabled,
+    metricsEnabled,
+    tracingEnabled,
+  } = init;
+
+  const methodOptions: Record<string, MethodDeploymentOptions> | undefined =
+    methodDeployments?.reduce((options, deployment) => {
+      const {
+        httpMethod,
+        publicPath,
+        proxyName,
+        cacheTtlInSeconds = 0,
+      } = deployment;
+
+      const path = proxyName
+        ? join(publicPath, `{${proxyName}+}`, httpMethod)
+        : publicPath === `/`
+        ? `//${httpMethod}`
+        : join(publicPath, httpMethod);
+
+      return {
+        ...options,
+        [path]: {
```

Comment:
don't understand the code fully, but can we do something like this with it?

```
        methodOptions: {
          '/bff/POST': {
            cachingEnabled: true,
            cacheTtl: cdk.Duration.minutes(5)
          },
          '/entry/{id}/GET': {
            cachingEnabled: false
          },
          '/pds/*/GET': {
            cachingEnabled: true,
            cacheTtl: cdk.Duration.minutes(60)
          }
        }
```

Reply:
This is an internal function. Generally, AWS does not support `*` in paths. For that there are greedy paths like `{proxy+}`. Such kind of paths are supported. I will describe this in the documentation. It works similar to the current catchAll routes.